oidc-api/src/test/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandlerTest.java

```diff
@@ -173,430 +198,625 @@ public void setUp() {
 
-        inOrder.verify(auditService)
-                .submitAuditEvent(
-                        OidcAuditableEvent.AUTHORISATION_INITIATED,
-                        CLIENT_SESSION_ID,
-                        session.getSessionId(),
-                        CLIENT_ID.getValue(),
-                        AuditService.UNKNOWN,
-                        AuditService.UNKNOWN,
-                        "123.123.123.123",
-                        AuditService.UNKNOWN,
-                        PERSISTENT_SESSION_ID,
-                        pair("client-name", RP_CLIENT_NAME));
-    }
+            var response = makeDocAppHandlerRequest();
 
-    @Test
-    void shouldRedirectToLoginWhenIdentityIsPresentInVtr() {
-        withExistingSession(session);
-        when(userContext.getClientSession()).thenReturn(clientSession);
-        when(userContext.getSession()).thenReturn(session);
-        when(clientSession.getAuthRequestParams())
-                .thenReturn(
-                        generateAuthRequest(Optional.of(jsonArrayOf("P2.Cl.Cm"))).toParameters());
+            assertThat(response, hasStatus(400));
+            assertThat(
+                    response.getBody(),
+                    equalTo(OAuth2Error.INVALID_CLIENT.toJSONObject().toJSONString()));
+            verifyNoInteractions(configService);
+            verifyNoInteractions(requestObjectService);
+        }
 
-        APIGatewayProxyResponseEvent response =
-                makeHandlerRequest(withRequestEvent(buildRequestParams(Map.of("vtr", "P2.Cl.Cm"))));
-        URI uri = URI.create(response.getHeaders().get(ResponseHeaders.LOCATION));
+        @Test
+        void shouldReturn400WhenAuthorisationRequestCannotBeParsed() {
+            APIGatewayProxyRequestEvent event = new APIGatewayProxyRequestEvent();
+            event.setQueryStringParameters(
+                    Map.of(
+                            "client_id",
+                            CLIENT_ID.getValue(),
+                            "redirect_uri",
+                            REDIRECT_URI,
+                            "scope",
+                            SCOPE,
+                            "invalid_parameter",
+                            "nonsense",
+                            "state",
+                            STATE.getValue()));
+            event.setRequestContext(
+                    new ProxyRequestContext()
+                            .withIdentity(new RequestIdentity().withSourceIp("123.123.123.123")));
+
+            APIGatewayProxyResponseEvent response = makeHandlerRequest(event);
+
+            assertThat(response, hasStatus(302));
+            assertEquals(
+                    "https://localhost:8080?error=invalid_request&error_description=Invalid+request%3A+Missing+response_type+parameter&state="
+                            + STATE.getValue(),
+                    response.getHeaders().get(ResponseHeaders.LOCATION));
```

Comment:
Not sure where exactly this exists in this diff, but we should make sure the doc app requests contain only the following query params:

client id
request
response_mode (or response_type, can't remember exactly)